### ISSUE TITLE
connected/disconnected popup notifications are annoying

### ISSUE THREAD

**Issue by ec1oud** [pwr-Solaar/Solaar#1915]
**Information**
- Solaar version 1.1.7
- Distribution: Arch
- Kernel version (ex. `uname -srmo`): `5.14.6`

**Screenshots**
![solaaaaaaar](https://user-images.githubusercontent.com/2946740/206376080-911783b7-4d58-468a-8073-c88c4aded035.png)

<details>

```
Solaar version 1.1.7

Unifying Receiver
  Device path  : /dev/hidraw3
  USB id       : 046d:C52B
  Serial       : 4DDD9BE3
    Firmware   : 24.07.B0030
    Bootloader : 02.09
    Other      : AA.AC
  Has 1 paired device(s) out of a maximum of 6.
  Notifications: wireless, software present (0x000900)
  Device activity counters: 1=198

  1: Wireless Mouse MX Master 2S
     Device path  : /dev/hidraw4
     WPID         : 4069
     Codename     : MX Master 2S
     Kind         : mouse
     Protocol     : HID++ 4.5
     Polling rate : 8 ms (125Hz)
     Serial number: 1330E0AD
     Model ID:      B01940690000
     Unit ID:       99BBC62D
        Bootloader: BOT 56.01.B0006
          Firmware: MPM 12.01.B0006
          Firmware: MPM 12.01.B0006
             Other: 
     The power switch is located on the base.
     Supports 32 HID++ 2.0 features:
         0: ROOT                   {0000} V0     
         1: FEATURE SET            {0001} V0     
         2: DEVICE FW VERSION      {0003} V2     
            Firmware: Bootloader BOT 56.01.B0006 406916A9FC5401
            Firmware: Firmware MPM 12.01.B0006 B01916A9FC5401
            Firmware: Firmware MPM 12.01.B0006 406916A9FC5401
            Firmware: Other   
            Unit ID: 99BBC62D  Model ID: B01940690000  Transport IDs: {'btleid': 'B019', 'wpid': '4069'}
         3: DEVICE NAME            {0005} V0     
            Name: Wireless Mouse MX Master 2S
            Kind: mouse
         4: WIRELESS DEVICE STATUS {1D4B} V0     
         5: RESET                  {0020} V0     
         6: CRYPTO ID              {0021} V0     
         7: BATTERY STATUS         {1000} V0     
            Battery: 50%, discharging, next level 20%.
         8: CONFIG DEVICE PROPS    {1806} V0    internal, hidden 
         9: CHANGE HOST            {1814} V1     
            Change Host        : 1:zen
        10: REPROG CONTROLS V4     {1B04} V3     
            Key/Button Actions (saved): {Left Button:Left Click, Right Button:Right Click, Middle Button:Mouse Middle Button, Back Button:Mouse Back Button, Forward Button:Mouse Forward Button, Mouse Gesture Button:Smart Shift, Smart Shift:Mouse Middle Button}
            Key/Button Actions        : {Left Button:Left Click, Right Button:Right Click, Middle Button:Mouse Middle Button, Back Button:Mouse Back Button, Forward Button:Mouse Forward Button, Mouse Gesture Button:Smart Shift, Smart Shift:Mouse Middle Button}
            Key/Button Diversion (saved): {Middle Button:Regular, Back Button:Regular, Forward Button:Regular, Mouse Gesture Button:Regular, Smart Shift:Regular}
            Key/Button Diversion        : {Middle Button:Regular, Back Button:Regular, Forward Button:Regular, Mouse Gesture Button:Regular, Smart Shift:Regular}
        11: ADJUSTABLE DPI         {2201} V1     
            Sensitivity (DPI) (saved): 1000
            Sensitivity (DPI)        : 1000
        12: VERTICAL SCROLLING     {2100} V0     
            Roller type: 3G
            Ratchet per turn: 24
            Scroll lines: 0
        13: SMART SHIFT            {2110} V0     
            Scroll Wheel Ratcheted (saved): Freespinning
            Scroll Wheel Ratcheted        : Freespinning
            Scroll Wheel Ratchet Speed (saved): 0
            Scroll Wheel Ratchet Speed        : 1
        14: HIRES WHEEL            {2121} V0     
            Multiplier: 8
            Has invert: Normal wheel motion
            Has ratchet switch: Free wheel mode
            High resolution mode
            HID notification
            Scroll Wheel Direction (saved): False
            Scroll Wheel Direction        : False
            Scroll Wheel Resolution (saved): False
            Scroll Wheel Resolution        : True
            Scroll Wheel Diversion (saved): False
            Scroll Wheel Diversion        : False
        15: GESTURE 2              {6501} V0     
            Gestures (saved): {45: True, 46: True}
            Gestures        : {46: True, 45: True}
            Gestures Diversion (saved): {46: False}
            Gestures Diversion        : {46: False}
        16: DFUCONTROL SIGNED      {00C2} V0     
        17: unknown:1813           {1813} V0    internal, hidden 
        18: unknown:1830           {1830} V0    internal, hidden 
        19: unknown:1890           {1890} V0    internal, hidden 
        20: unknown:1891           {1891} V0    internal, hidden 
        21: unknown:18A1           {18A1} V0    internal, hidden 
        22: unknown:18C0           {18C0} V0    internal, hidden 
        23: unknown:1DF3           {1DF3} V0    internal, hidden 
        24: unknown:1E00           {1E00} V0    hidden 
        25: unknown:1EB0           {1EB0} V0    internal, hidden 
        26: unknown:1803           {1803} V0    internal, hidden 
        27: unknown:1861           {1861} V0    internal, hidden 
        28: unknown:9001           {9001} V0    internal, hidden 
        29: unknown:9200           {9200} V0    internal, hidden 
        30: unknown:9202           {9202} V0    internal, hidden 
        31: OOBSTATE               {1805} V0    internal, hidden 
     Has 8 reprogrammable keys:
         0: Left Button               , default: Left Click                  => Left Click                
             mse, pos:0, group:1, group mask:g1
             reporting: default
         1: Right Button              , default: Right Click                 => Right Click               
             mse, pos:0, group:1, group mask:g1
             reporting: default
         2: Middle Button             , default: Mouse Middle Button         => Mouse Middle Button       
             mse, reprogrammable, divertable, raw XY, pos:0, group:3, group mask:g1,g2,g3
             reporting: default
         3: Back Button               , default: Mouse Back Button           => Mouse Back Button         
             mse, reprogrammable, divertable, raw XY, pos:0, group:2, group mask:g1,g2
             reporting: default
         4: Forward Button            , default: Mouse Forward Button        => Mouse Forward Button      
             mse, reprogrammable, divertable, raw XY, pos:0, group:2, group mask:g1,g2
             reporting: default
         5: Mouse Gesture Button      , default: Gesture Button Navigation   => Smart Shift               
             mse, reprogrammable, divertable, raw XY, pos:0, group:3, group mask:g1,g2,g3
             reporting: default
         6: Smart Shift               , default: Smart Shift                 => Mouse Middle Button       
             mse, reprogrammable, divertable, raw XY, pos:0, group:3, group mask:g1,g2,g3
             reporting: default
         7: Virtual Gesture Button    , default: Virtual Gesture Button      => Virtual Gesture Button    
             divertable, virtual, raw XY, force raw XY, pos:0, group:4, group mask:empty
             reporting: default
     Has 2 gesture(s), 0 param(s) and 4 spec(s):
        Thumbwheel                 Enabled(   0): True   Diverted:(   0) False
        NaturalScrolling           Enabled(   1): True   Diverted:(None) None
        period unit                Spec   (   3): 800
        DVI field width            Spec   (   1): 8
        field widths               Spec   (   2): 8
        resolution                 Spec   (   4): 2048
     Battery: 50%, discharging, next level 20%.
```

- Contents of `~/.config/solaar/config.yaml` (or `~/.config/solaar/config.json` if `~/.config/solaar/config.yaml` not present):

```
- 1.1.7
- _NAME: Wireless Mouse MX Master 2S
  _absent: [hi-res-scroll, lowres-smooth-scroll, thumb-scroll-mode, thumb-scroll-invert, onboard_profiles, report_rate, pointer_speed, speed-change, backlight,
    backlight-timed, fn-swap, persistent-remappable-keys, disable-keyboard-keys, divert-crown, crown-smooth, divert-gkeys, m-key-leds, mr-key-led, multiplatform,
    gesture2-params, lowres-scroll-mode, sidetone, equalizer]
  _battery: 4096
  _modelId: B01940690000
  _sensitive: {change-host: true, hires-smooth-resolution: ignore, reprogrammable-keys: true, smart-shift: true}
  _serial: 1330E0AD
  _unitId: 99BBC62D
  _wpid: '4069'
  change-host: null
  divert-keys: {82: 0, 83: 0, 86: 0, 195: 0, 196: 0}
  dpi: 1000
  gesture2-divert: {46: false}
  gesture2-gestures: {45: true, 46: true}
  hires-scroll-mode: false
  hires-smooth-invert: false
  hires-smooth-resolution: false
  reprogrammable-keys: {80: 80, 81: 81, 82: 82, 83: 83, 86: 86, 195: 196, 196: 82}
  scroll-ratchet: 1
  smart-shift: 0
```

- Errors or warrnings from Solaar:

**Describe the bug**
This is rather like #335 but that one is marked closed already.

When I switch the mouse between machines, if a machine is running solaar and the mouse connects via bluetooth, solaar pops up a dialog each time the mouse connects or disconnects via bluetooth.  Not only are these annoying in the first place, but the dialog does not have a Close button (except the one on the titlebar), sometimes it's FROZEN so I can't close it right away, and it never closes by itself.  Over time, my desktop gets littered with these dialogs, which I tend to just leave and forget about, by raising other windows on top, because they are extra trouble to close.  A ridiculous situation, IMO.

I'm running openbox, so the workarounds other users have, to disable certain classes of notifications system-wide, are also more trouble to figure out in my case.

Usability-wise,
1. solaar NEEDS a setting to turn these off!!!  Regardless whatever other workarounds you suggest, this has to be there. These notifications are of dubious utility to anyone.  But I don't see a preferences or settings item AT ALL on the tray icon menu?
2. every notification should be done via the standard desktop notification system, if possible.  (I guess you do that, but I'm not running dunst at this moment, although I can do that.)
3. failing that: any throwaway popup dialog ought to have a big Close button at the bottom.  It's very easy for me to miss while trying to click the one on the titlebar, often causing another window (the one I was actually trying to work with, when this interruption happens) to raise on top, and at that point it's easier to just let out another curse for Solaar and move on, and leave the popup there for the ages.
4. why not just close it automatically after a few seconds?
5. if you're not going to close it automatically, why not re-use existing notification windows to show the latest notification? The mouse cannot be both connected and disconnected at the same time!  Even if this is Schroedinger's Mouse, it cannot be *connected and disconnected several times in a row at the same time*, on different virtual desktops!
6. a graphical UI should never FREEZE! (maybe Python is the problem? can you have threads to isolate operations that might hang for a while?)
7. don't write long-running, core system utilities in Python.  I prefer to run real native executables: they take less memory and CPU time.  Python is a scripting language, for getting short-running tasks done easily.

**To Reproduce**
1. switch the mouse back and forth between machines using the button on the bottom.

**Additional context**
The mouse can connect to 3 different machines via the button on the bottom.  Each "slot" can be assigned to a Bluetooth address, or a Logitech Unifying Receiver.  This problem occurs only with Bluetooth connections.  But when you are connected to machine 2, say, and you want to connect to machine 1, you have to cycle through 3 first, to get back to 1.  That means you could be getting these infernal useless dialogs popping up on multiple machines at the same time!
</details>

**Comment by ec1oud** [comment]
oryband's comment to configure dunst to hide all notifications from solaar (and run dunst!) is helpful:
```
[solaar]
   appname = "Solaar"
   format = ""
```

**Comment by pfps** [comment]
Solaar's pop-up notifications use standard mechanisms - Notify in Glib.  On well-behaved window systems they go away on their own.  Good window systems should have a mechanism for not showing Solaar pop-up notifications.

**Comment by pfps** [comment]
Further, are you sure that Solaar is the system that is creating these notifications?  On my system I see pop-up notifications for device connection and disconnection even when Solaar is not running.  There are no "Disconnected" or "Connected" strings in the current version of Solaar.

**Comment by epilogic-ch** [comment]
I came across this topic while Googling the same kind of problem. At first, I thought the problem was with Solaar, but in reality, the “Connected” / “Disconnected” notifications are generated by Bluetooth.

For those experiencing the same problem, I was able to avoid these excessive notifications by disabling them in the Bluetooth settings:

1. Open Bluetooth Manager
2. Menu View > Plugins
3. Disable (uncheck) “ConnectionNotifier”
